src/main/java/com/questhelper/QuestHelperPlugin.java

```diff
@@ -310,21 +314,19 @@ public void updateQuestList()
 	{
 		if (client.getGameState() == GameState.LOGGED_IN)
 		{
-			List<QuestHelper> questHelpers = new ArrayList<>(quests.values());
-
-			if (config.filterListBy() == QuestHelperConfig.QuestFilter.HIDE_DONE)
-			{
-				questHelpers = quests.values().stream().filter(quest -> !quest.isCompleted()).collect(Collectors.toList());
-			}
-			else if (config.filterListBy() == QuestHelperConfig.QuestFilter.HIDE_LACKING_REQ)
-			{
-				questHelpers =
-					quests.values().stream().filter(quest -> quest.hasRequirements() && !quest.isCompleted()).collect(Collectors.toList());
-			}
-
-			List<QuestHelper> finalQuestHelpers = questHelpers;
+			List<QuestHelper> allQuests = new ArrayList<>(quests.values());
+
+			List<QuestHelper> filteredQuests = allQuests
+				.stream()
+				.filter(q -> config.filterListBy().test(q, config))
+				.filter(q -> (config.showCompletedQuests() && q.isCompleted()) || !q.isCompleted())
+				.sorted(config.orderListBy().getComparator())
+				.collect(Collectors.toList());
+			Map<QuestHelperQuest, QuestState> completedQuests = quests.values()
+				.stream()
+				.collect(Collectors.toMap(QuestHelper::getQuest, q -> q.getQuest().getState(client)));
```

Comment:
This is a bit problematic for quests/miniquests with more complex definitions for whether they're started/completed (RFD sub-quests, if we add non-listed tasks such as Barb Training) as the checks don't extend to them (https://github.com/RuneStar/cs2-scripts/blob/e0bce7a2bc1e27a2910861a9e18140207587508b/scripts/%5Bproc%2Cquestlist_progress%5D.cs2). 

May be worth in QuestHelperPlugin having:

```
.collect(Collectors.toMap(QuestHelper::getQuest, q -> q.getQuest().getState(client)));
```

Instead be:

```
.collect(Collectors.toMap(QuestHelper::getQuest, q -> q.getState(client)));
```

Where `getState` by default calls getState of the quest as done above, but can be overriden. I believe the RFD sub-sections already have isComplete overriden in pretty much an identical way you'd want this to be, so should be easy to do. Would then also make this easily scalable in the future.

Reply:
When I get back home I'll change this